Documentation/coding-guidelines/breaking-change-rules.md

```diff
@@ -142,6 +138,9 @@ Breaking Change Rules
 * Removing the implementation of an interface on a type  
 > It is not breaking when you added the implementation of an interface which derives from the removed interface. For example, you removed `IDisposable`, but implemented `IComponent`, which derives from `IDisposable`.
 
+* Adding an interface implementation to a type
+> This can be a compile-time breaking change. Consider a type `public class MyType : IEnumerable<A>`. A developer can call the standard `ToList()` extension method and get a `List<A>` as a result. If the class is ever extended in the future as `public class MyType : IEnumerable<A>, IEnumerable<B>`, the call to `ToList()` will fail to compile since type inference cannot determine if the return type should be `List<A>` or `List<B>`. Type inference issues aside, adding interfaces like `ISerializable` affects the ability of a designer or serializer to generate code in a downlevel-compatible fashion.
```

Comment:
We have added an interface implementation to a type many times, without issues.

The cases you mention should be something one should consider as an exception to the rule. They are not a good reason to disallow this wholesale.

If we were holding a hardline on the potential compile-time breaking changes, we would not be able to add *anything* to the framework.

Reply:
Another example is adding an interface, such as IFormattable, to an existing type; this would cause existing code to behave differently than before such as an existing call to ```String.Format("{0:blah}", x);``` - that would previously silently pass on malformed ```"blah"``` data, would now call your ```IFormattable.ToString("blah", ci);``` and possibly throw or have new behavior than before.